File: sdk/go/tenants.go
package qeetid

import (
	"context"
	"net/http"
	"net/url"
	"strconv"
)

type Tenant struct {
	ID        string `json:"id"`
	Name      string `json:"name"`
	Slug      string `json:"slug"`
	Region    string `json:"region,omitempty"`
	CreatedAt string `json:"created_at"`
	UpdatedAt string `json:"updated_at,omitempty"`
}

type CreateTenantInput struct {
	Name   string `json:"name"`
	Slug   string `json:"slug"`
	Region string `json:"region,omitempty"`
}

type UpdateTenantInput struct {
	Name   *string `json:"name,omitempty"`
	Region *string `json:"region,omitempty"`
}

type TenantPage struct {
	Data       []Tenant
	NextCursor string
}

type Tenants struct{ http *httpClient }

func (t *Tenants) Create(ctx context.Context, in CreateTenantInput) (*Tenant, error) {
	var out Tenant
	err := t.http.do(ctx, http.MethodPost, "/v1/tenants", nil, in, &out, false)
	return &out, err
}

func (t *Tenants) Get(ctx context.Context, id string) (*Tenant, error) {
	var out Tenant
	err := t.http.do(ctx, http.MethodGet, "/v1/tenants/"+url.PathEscape(id), nil, nil, &out, true)
	return &out, err
}

func (t *Tenants) Update(ctx context.Context, id string, in UpdateTenantInput) (*Tenant, error) {
	var out Tenant
	err := t.http.do(ctx, http.MethodPatch, "/v1/tenants/"+url.PathEscape(id), nil, in, &out, false)
	return &out, err
}

func (t *Tenants) Delete(ctx context.Context, id string) error {
	return t.http.do(ctx, http.MethodDelete, "/v1/tenants/"+url.PathEscape(id), nil, nil, nil, true)
}

func (t *Tenants) List(ctx context.Context, limit int, cursor string) (*TenantPage, error) {
	q := url.Values{}
	if limit > 0 {
		q.Set("limit", strconv.Itoa(limit))
	}
	if cursor != "" {
		q.Set("cursor", cursor)
	}
	var env struct {
		Items      []Tenant `json:"items"`
		Data       []Tenant `json:"data"`
		NextCursor string   `json:"next_cursor"`
	}
	if err := t.http.do(ctx, http.MethodGet, "/v1/tenants", q, nil, &env, true); err != nil {
		return nil, err
	}
	items := env.Items
	if items == nil {
		items = env.Data
	}
	return &TenantPage{Data: items, NextCursor: env.NextCursor}, nil
}

File: sdk/js/node/src/tenants.ts
import type { HttpClient } from "./client.js";
import type { ListParams, Page } from "./users.js";

export interface Tenant {
  id: string;
  name: string;
  slug: string;
  region?: string;
  created_at: string;
  updated_at?: string;
}

export interface CreateTenantInput {
  name: string;
  slug: string;
  region?: string;
}

export interface UpdateTenantInput {
  name?: string;
  region?: string;
}

interface ListEnvelope {
  items?: Tenant[];
  data?: Tenant[];
  next_cursor?: string;
}

export class Tenants {
  constructor(private readonly http: HttpClient) {}

  create(input: CreateTenantInput): Promise<Tenant> {
    return this.http.post<Tenant>("/v1/tenants", input);
  }

  get(id: string): Promise<Tenant> {
    return this.http.get<Tenant>(`/v1/tenants/${encodeURIComponent(id)}`);
  }

  update(id: string, input: UpdateTenantInput): Promise<Tenant> {
    return this.http.patch<Tenant>(`/v1/tenants/${encodeURIComponent(id)}`, input);
  }

  delete(id: string): Promise<void> {
    return this.http.delete<void>(`/v1/tenants/${encodeURIComponent(id)}`);
  }

  async list(params: Omit<ListParams, "tenant"> = {}): Promise<Page<Tenant>> {
    const res = await this.http.get<ListEnvelope>("/v1/tenants", {
      query: { limit: params.limit, cursor: params.cursor },
    });
    return { data: res.items ?? res.data ?? [], nextCursor: res.next_cursor };
  }
}
